File: LeetCode/ConstructAlgorithm/Js/Leetcode667-beautiful-arrangement-ii.js
var constructArray = function(n, k) {
    let ans = [];
    for(let i = 1;i < n - k;i++) ans.push(i);
    for(let [i,j] = [n - k,n];i <= j;i++,j--){
        ans.push(i);
        if(i != j) ans.push(j);
    }
    return ans;
};

File: LeetCode/ConstructAlgorithm/Golang/Leetcode667-beautiful-arrangement-ii.go
package sbgo

func constructArray(n int, k int) []int {
    ans := []int{}
    for i := 1;i < n - k;i++ {
        ans = append(ans,i)
    }
    for i,j := n - k,n;i <= j;i,j = i + 1,j - 1 {
        ans = append(ans,i)
        if i != j {
            ans = append(ans,j)
        }
    }
    return ans
}
